torchtune/models/mistral/modules/transformer.py

```diff
@@ -0,0 +1,258 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Optional, Tuple
+
+import torch
+from torch import nn, Tensor
+
+from torchtune.modules import TransformerDecoderLayer
+from torchtune.modules.kv_cache import KVCache
+from torchtune.modules.transformer import _get_clones
+
+
+class TransformerDecoderWithHiddenOutput(nn.Module):
```

Comment:
Personally I find the name a bit confusing, to me it implies that we should be able to access the hidden layers which we actually can't (I know there is other work ongoing with the transformer decoder refactor so if this is covered there feel free to ignore)

Reply:
yep agreed - i think for the most part you can ignore my transformerdecoder work here since it'll be removed once I merge in the transformerdecoder refactor.